source/slang/hlsl.meta.slang

```diff
@@ -1438,8 +1503,8 @@ void InterlockedXor(__ref uint dest, uint value, out uint original_value);
 
 __generic<T : __BuiltinFloatingPointType>
 __target_intrinsic(hlsl)
-__target_intrinsic(cpu)
-__target_intrinsic(cuda)
+__target_intrinsic(cuda, "$P_isfinite($0)")
```

Comment:
Well, we already do have the mechanism where `__target_intrinsic(someTarget)` is sugar for `__target_intrinsic(someTarget, "theNameOfTheSlangFunction")`, so the original idea (which made sense for the HLSL and GLSL targets) was that the 99% case would be that a builtin would have the exact same name in the target language as in Slang.

The prefixing we are doing for CPU/CUDA functions means that the common case for them is quite different, and we end up having to do the explicit `$P_theNameOfTheSlangFunction` all over the place. *If* we are okay with relying on C++ overload resolution for the generated CPU/CUDA code then maybe we could consider walking this back and just using the bare function name again.

In our conversation about the prefixing, you mentioned that it was helpful in making sure we didn't accidentally introduce performance pitfalls (e.g., calling `sin()` instead of `sinf()` on a `float` will Just Work, but it means you now have `double`-precision math you might not have wanted).

One alternative would be to make the prefixing be built-in/default behavior for the CUDA/CPU targets (but not for HLSL/GLSL) so that most of the functions could just use `__target_intrinsic(cuda)`, etc.